{{cookiecutter.project_name}}/docs/developer_docs.md

```diff
@@ -53,11 +93,258 @@ The following pre-commit checks for errors, inconsistencies and typing.
+    config files, such as `.prettierrc.yml` for **prettier** and `.flake8` for **flake8**.
+
+### API design
+
+Scverse ecosystem packages should operate on [AnnData][] and/or [MuData][] datastructures and typically use an API
+as originally [introduced by scanpy][scanpy-api] with the following submodules:
+
+-   `pp` for preprocessing
+-   `tl` for tools (that, compared to `pp` generate interpretable output, often associated with a corresponding plotting
+    function)
+-   `pl` for plotting functions
+
+You may add additional submodules as appropriate. While we encourage to follow a scanpy-like API for ecosystem packages,
+there may also be good reasons to choose a different approach, e.g. using an object-oriented API.
+
+[scanpy-api]: https://scanpy.readthedocs.io/en/stable/usage-principles.html
+
+### Ignore certain lint warnings
+
+The [pre-commit checks](#pre-commit-checks) include [flake8](https://flake8.pycqa.org/en/latest/) which checks
+for errors in Python files, including stylistic errors.
+
+In some cases it might overshoot and you may have good reasons to ignore certain warnings.
+
+To ignore an specific error on a per-case basis, you can add a comment `# noqa` to the offending line. You can also
+specify the error ID to ignore, with e.g. `# noqa: E731`. Check the [flake8 guide][] for reference.
+
+Alternatively, you can disable certain error messages for the entire project. To do so, edit the `.flake8`
+file in the root of the repository. Add one line per linting code you wish to ignore and don't forget to add a comment.
+
+```toml
+...
+# line break before a binary operator -> black does not adhere to PEP8
+W503
+# line break occured after a binary operator -> black does not adhere to PEP8
+W504
+...
+```
+
+[flake8 guide]: https://flake8.pycqa.org/en/3.1.1/user/ignoring-errors.html
+
+### Using VCS-based versioning
+
+By default, the template uses hard-coded version numbers that are set in `pyproject.toml` and [managed with
+bump2version](#making-a-release). If you prefer to have your project automatically infer version numbers from git
+tags, it is straightforward to switch to vcs-based versioning using [hatch-vcs][].
+
+In `pyproject.toml` add the following changes, and you are good to go!
+
+```diff
+--- a/pyproject.toml
++++ b/pyproject.toml
+@@ -1,11 +1,11 @@
+ [build-system]
+ build-backend = "hatchling.build"
+-requires = ["hatchling"]
++requires = ["hatchling", "hatch-vcs"]
+
+
+ [project]
+ name = "{{ cookiecutter.project_name }}"
+-version = "0.3.1dev"
++dynamic = ["version"]
+
+@@ -60,6 +60,9 @@
++[tool.hatch.version]
++source = "vcs"
++
+ [tool.coverage.run]
+ source = ["{{ cookiecutter.project_name }}"]
+ omit = [
+```
+
+Don't forget to update the [Making a release section](#making-a-release) in this document accordingly, after you are done!
+
+[hatch-vcs]: https://pypi.org/project/hatch-vcs/
+
+### Template sync with _cookietemple_
```

Comment:
This is going to change a lot of things. This template will move somewhere else and the documentation will be hosted at another place etc. I'd remove this section for now and I will suggest the changes then when it's ready.